src/posit/connect/resources.py

```diff
@@ -119,65 +128,61 @@ def fetch(self) -> List[T]:
         List[T]
         """
         endpoint = self._ctx.url + self._path
-        response = self._ctx.session.get(endpoint)
+        response = self._ctx.session.get(endpoint, params=conditions)
         results = response.json()
         return [self._to_instance(result) for result in results]
 
-    def reload(self) -> Self:
-        """Reloads the collection from Connect.
-
-        Returns
-        -------
-        Self
-        """
-        self._cache = None
-        return self
-
     def _to_instance(self, result: dict) -> T:
         """Converts a result into an instance of T."""
         uid = result[self._uid]
         path = posixpath.join(self._path, uid)
         return self._create_instance(path, **result)
 
-    @property
-    def _data(self) -> List[T]:
-        """Get the collection.
-
-        Fetches the collection from Connect and caches the result. Subsequent invocations return the cached results unless the cache is explicitly reset.
-
-        Returns
-        -------
-        List[T]
-
-        See Also
-        --------
-        cached
-        reload
-        """
-        if self._cache is None:
-            self._cache = self.fetch()
-        return self._cache
-
     @overload
     def __getitem__(self, index: int) -> T: ...
 
     @overload
     def __getitem__(self, index: slice) -> Sequence[T]: ...
 
-    def __getitem__(self, index):
-        return self._data[index]
+    def __getitem__(self, index) -> Sequence[T] | T:
+        data = self.fetch()
+
+        if isinstance(index, int):
+            if index < 0:
+                # Handle negative indexing
+                data = list(data)
+                return data[index]
+            for i, value in enumerate(data):
+                if i == index:
+                    return value
+            raise KeyError(f"Index {index} is out of range.")
+
+        if isinstance(index, slice):
+            # Handle slicing with islice
+            start = index.start or 0
+            stop = index.stop
+            step = index.step or 1
+            if step == 0:
+                raise ValueError("slice step cannot be zero")
+            return [
+                value
+                for i, value in enumerate(islice(data, start, stop))
+                if (i + start) % step == 0
+            ]
+
+        raise TypeError(f"Index must be int or slice, not {type(index).__name__}.")
 
     def __len__(self) -> int:
-        return len(self._data)
+        return len(list(self.fetch()))
```

Comment:
Should these methods cache the result? I'm concerned about a client that writes
```
def do_something(items):
  for i in range(len(items)):
     # do something with items[i]

do_something(client.packages)
```

Reply:
We've been going back and forth on caching. I actually think I should revert these changes back to using the `_data` property.